app-frontend/src/app/components/lab/diagramNodeHeader/diagramNodeHeader.html

```diff
@@ -28,8 +28,36 @@
             ng-if="$ctrl.node.apply"></span>
       {{$ctrl.typeMap[$ctrl.node.type]}}
     </div>
-    <div class="node-header-label">
-      {{$ctrl.node.metadata.label}}
+    <div>
+      <span ng-if="!$ctrl.isEditNodeLabel">
+        <button class="btn node-button pate-title-edit detail-title-edit"
```

Comment:
I think button should be after the title so that the two rows will have their text aligned.

Reply:
Additionally, we'll want to add a new class to this button to shrink the font size a little bit.

Reply:
And then we'll want to add a `margin-right: 0.25em` to the `node-header-label` class.